### PR TITLE
boards: arm: add PhyBOARD Polis (i.MX8MM)

### PR DESCRIPTION
**Pull request by pefech**
Add board definitions to run Zephyr on the M4-Core
of the i.MX8M Mini SoC on the PhyBOARD Polis.
You can find more about this board here: [PhyBOARD Polis](https://www.phytec.de/produkte/single-board-computer/phyboard-polis-imx8m-mini/) or [PhyBOARD Polis (Phytec USA)](https://www.phytec.com/product/phyboard-i-mx-8m-mini-polis/)

All configured devices and upload methods were tested.

Signed-off-by: Peter Fecher <p.fecher@phytec.de>

### CODE REVIEW

**Review by galak — DISMISSED** [review]
Please reduce image file size 5MB is tooooo big ;).

**Comment by pefech** [comment]
Oh, I didn't saw that - fixed.

**Comment by pefech** [comment]
fixed typo

**Comment by mmahadevan108** [comment]
Please add the board details to the `CODEOWNERS` file.

**Comment by pefech** [comment]
Added to ```CODEOWNERS``` file with last push.

**Comment by mmahadevan108** [comment]
@galak, can you please have a review again.

File: boards/arm/mimx8mm_phyboard_polis/mimx8mm_phyboard_polis.dts
Comment (danieldegrasse) [inline boards/arm/mimx8mm_phyboard_polis/mimx8mm_phyboard_polis.dts:34]: Why is `pinctrl` needed here? The `gpio_pin_configure` API should be able to select GPIO mode when it is used to configure pins. If this isn't working, please feel free to open an issue against this SOC

File: boards/arm/mimx8mm_phyboard_polis/mimx8mm_phyboard_polis_defconfig
Comment (danieldegrasse) [inline boards/arm/mimx8mm_phyboard_polis/mimx8mm_phyboard_polis_defconfig:24]: Need to add a newline at the end of this file

File: boards/arm/mimx8mm_phyboard_polis/doc/index.rst
Comment (pefech) [reply boards/arm/mimx8mm_phyboard_polis/doc/index.rst:306]: ```west debug``` also works fine. But I prefer the way I documented. If you think it's better to have it documented with ```west debug```, I can change it.

File: boards/arm/mimx8mm_phyboard_polis/mimx8mm_phyboard_polis.dts
Comment (pefech) [reply boards/arm/mimx8mm_phyboard_polis/mimx8mm_phyboard_polis.dts:34]: You are completely right. I'll remove the ```pinctrl``` for the leds.

File: boards/arm/mimx8mm_phyboard_polis/doc/index.rst
Comment (danieldegrasse) [reply boards/arm/mimx8mm_phyboard_polis/doc/index.rst:306]: I'm not opposed to the documentation here, but I would add a note clarifying that `west debug` or `west debugserver` are both supported, just to avoid confusing users

File: boards/arm/mimx8mm_phyboard_polis/mimx8mm_phyboard_polis_defconfig
Comment (danieldegrasse) [inline boards/arm/mimx8mm_phyboard_polis/mimx8mm_phyboard_polis_defconfig:20]: If these KConfigs are only needed when using remoteproc, could you add them to the board's `KConfig.defconfig`, and make them depend on a remoteproc related KConfig value? I'd prefer not to require OpenAMP be enabled for this board.

File: boards/arm/mimx8mm_phyboard_polis/mimx8mm_phyboard_polis_defconfig
Comment (danieldegrasse) [inline boards/arm/mimx8mm_phyboard_polis/mimx8mm_phyboard_polis_defconfig:24]: You have defaults in the board `Kconfig.defconfig` for when XIP is disabled, but are requiring it be enabled here. Does this board support running with `CONFIG_XIP=n`?

**Comment by dleach02** [comment]
@pefech will you be returning to this PR?

**Comment by pefech** [comment]
My priority was on other tasks. I'll continue to work on this PR soon.

**Comment by MaureenHelm** [comment]
@danieldegrasse please take another look

File: boards/arm/mimx8mm_phyboard_polis/doc/index.rst
Comment (danieldegrasse) [inline boards/arm/mimx8mm_phyboard_polis/doc/index.rst:105]: Nit: can you fix spelling here?

**Review by danieldegrasse — APPROVED** [review]

**Comment by pefech** [comment]
Is there any chance someone has some time to make the second review on this?

**Review by carlocaione — APPROVED** [review]

**Pull request merged**
